services/orderService.js

```diff
@@ -0,0 +1,85 @@
+const yup = require('yup');
+const find = require('lodash/find');
+const productModel = require('../models/productModel');
+const orderModel = require('../models/orderModel');
+const constants = require('../constants/constants');
+
+const getAll = async (req, res, next) => {
+  let data;
+  try {
+    data = await orderModel.getAll();
+    res.status(200).send(data);
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const getOrderById = async (req, res, next) => {
+  let data;
+  try {
+    data = await orderModel.getOrderById(req.params.id);
+    res.status(200).send(data);
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const create = async (req, res, next) => {
+  const order = {
+    ...req.body,
+  };
+
+  // validation of schema for necessary fields
+  const schema = yup.object().shape({
+    shipment_method: yup.string(),
+    products: yup
+      .array()
+      .of(
+        yup.object().shape({
+          id: yup.number().required(),
+          quantity: yup.number().required(),
+        }),
+      )
+      .required(),
+    customer_id: yup.number().required(),
+  });
+
+  schema
+    .validate(order)
+    .then(async (validatedOrder) => {
+      // fetch product data from db - ensure user cant pass in their own prices
+      const products = await productModel.getProductsByIds(
+        validatedOrder.products.map((product) => product.id),
+      );
+      // iterate and calculate product prices
+      let total_before_tax = 0;
+      validatedOrder.products.forEach((product) => {
+        const productData = find(
+          products,
+          (productDbData) => productDbData.id === product.id,
+        );
```

Comment:
I was trying to understand how this function works. I found find() method that can be called on arrays: https://www.w3schools.com/jsref/jsref_find.asp

I assume this is pretty much the same except if not calling directly on an array, the syntax is: `find(_array_, _searchFunction_)` ?

Reply:
This is actually the [lodash find](https://lodash.com/docs/4.17.15#find) - I suppose `Array.find` would've worked just as well

Reply:
Ah! I missed the import at the top.